src/inference.jl

```diff
@@ -621,10 +647,17 @@ function inference(;
         fe_values        = !isnothing(fe_actor) ? score_snapshot_iterations(fe_actor) : nothing
 
         inference_invoke_callback(callbacks, :after_inference, fmodel)
+        
 
         return InferenceResult(posterior_values, fe_values, fmodel, freturval)
     catch error
-        __inference_process_error(error)
+        if allow_failed
```

Comment:
Please do not duplicate code, just move the existing lines of code outside of the try-catch block